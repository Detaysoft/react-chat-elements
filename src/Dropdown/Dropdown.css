.rce-dropdown-container {
    position: relative;
}
.rce-dropdown {
    min-width: 100%;
    box-sizing: border-box;
    padding: 8px 15px;
    background: #fff;
    border-radius: 5px;
    display: none;
    flex-direction: row;
    align-items: center;
    justify-content: center;
    transform: scale(0);
    position: absolute;
    box-shadow: 0px 0px 5px 0px rgba(163, 163, 163, 1);
    transform-origin: left top;
    z-index: 99999;
}

.rce-dropdown.dropdown-show{
    animation: dropdown-scaling 0.2s ease forwards;
    display: flex;
}

@keyframes dropdown-scaling {
    0% {
        opacity: 0;
    }
    50% {
        opacity: 0.5;
        transform: scale(1.1);
    }
    100% {
        opacity: 1;
        transform: scale(1);
    }
}

.rce-dropdown.dropdown-hide{
    animation: dropdown-reverse-scaling 0.2s ease forwards;
    display: flex;
}

@keyframes dropdown-reverse-scaling {
    0% {
        opacity: 1;
        transform: scale(1);
    }
    50% {
        opacity: 0.5;
        transform: scale(1.1);
    }
    100% {
        opacity: 0;
        transform: scale(0);
    }
}

.rce-dropdown-open__nortwest {
    transform-origin: left top;
    left: 0;
    top: 100%;
    margin-top: 5px
}

.rce-dropdown-open__norteast {
    transform-origin: right top;
    right: 0;
    top: 100%;
    margin-top: 5px
}

.rce-dropdown-open__southwest {
    transform-origin: left bottom;
    left: 0;
    bottom: 100%;
    margin-bottom: 5px
}

.rce-dropdown-open__southeast {
    transform-origin: right bottom;
    right: 0;
    bottom: 100%;
    margin-bottom: 5px
}

.rce-dropdown ul {
    list-style: none;
    margin: 0;
    padding: 0;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

.rce-dropdown ul li {
    white-space: nowrap;
    color: #767676;
    padding: 8px;
    cursor: pointer;
    font-size: 16px;
    width: 100%;
    border-bottom: 1px solid #e9e9e9;
    box-sizing: border-box;
    user-select: none;
}

.rce-dropdown ul li:last-child {
    border: none;
}

.rce-dropdown ul li:hover a {
    color: #3a6d8c;
}

.rce-dropdown.fade {
    opacity: 0;
    transform: scale(1);
    animation: dropdown-fade 0.5s ease forwards;
}

@keyframes dropdown-fade {
    0% {
        opacity: 0;
    }
    100% {
        opacity: 1;
    }
}
