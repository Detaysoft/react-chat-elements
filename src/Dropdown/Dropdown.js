import React, { Component } from 'react';
import './Dropdown.css';

import Button from '../Button/Button';

const classNames = require('classnames');

export class Dropdown extends Component {
    constructor(props) {
        super(props);

        this.state = {
            show: undefined,
        };
    }

    onBlur(e) {
        if (this.state.show === true)
            this.setState({show: false});
    }

    render() {
        return (
            <div className={classNames('rce-dropdown-container', this.props.className)} onBlur={this.onBlur.bind(this)}>
                {
                    <Button
                        {...this.props.buttonProps}
                        onClick={() => this.setState({ show: !this.state.show })} />
                }
                <div
                    className={classNames(
                        'rce-dropdown',
                        this.props.animationType,
                        'rce-dropdown-open__' + this.props.animationPosition,
                        { 'dropdown-hide': this.state.show === false },
                        { 'dropdown-show': this.state.show === true })}>

                        <ul>
                            {
                                this.props.items.map((x, i) => (
                                    <li key={i} onMouseDown={(e) => this.props.onSelect(i)}>
                                        <a>{x}</a>
                                    </li>
                                ))
                            }
                        </ul>
                </div>
            </div>
        );
    }
}

Dropdown.defaultProps = {
    animationType: 'default',
    animationPosition: 'nortwest',
    items: [],
    onSelect: Function,
    buttonProps: null
};

export default Dropdown;
