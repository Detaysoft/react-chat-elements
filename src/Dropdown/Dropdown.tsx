import React, { useState } from 'react'
import './Dropdown.css'
import classNames from 'classnames'
import Button from '../Button/Button'
import { IDropdownProps, IDropdownItemType } from '../type'

const Dropdown: React.FC<IDropdownProps> = ({ animationPosition = 'nortwest', animationType = 'default', ...props }) => {
  const [show, setShow] = useState<boolean | undefined>(undefined)

  const onBlur = () => {
    if (show === true) setShow(false)
  }

  return (
    <div className={classNames('rce-dropdown-container' as any, props.className as any)} onBlur={onBlur}>
      {<Button {...props.buttonProps} onClick={() => setShow(!show)} />}
      <div
        className={classNames(
          'rce-dropdown' as any,
          animationType as any,
          'rce-dropdown-open__' + animationPosition as any,
          { 'dropdown-hide': show === false } as any,
          { 'dropdown-show': show === true } as any
        )}
      >
        <ul>
          {props.title && <span className='rce-dropdown-title'>{props.title}</span>}
          {props.items?.map((x: IDropdownItemType, i: number) => (
            <li key={i} onMouseDown={e => props.onSelect(i)}>
              {x instanceof Object ? (
                  "icon" in x && x.icon ? (
                  <span className='rce-button-icon--container'>
                    {(x.icon.float === 'right' || !x.icon.float) && <a>{x.text}</a>}

                    <span
                      style={{ float: x.icon.float, color: x.icon.color, fontSize: x.icon.size || 12 }}
                      className={classNames('rce-button-icon' as any, x.icon.className as any)}
                    >
                      {x.icon.component}
                    </span>

                    {x.icon.float === 'left' && <a>{x.text}</a>}
                  </span>
                ) : (
                  <a>{"text" in x ? x.text : null}</a>
                )
              ) : (
                <a>{x}</a>
              )}
            </li>
          ))}
        </ul>
      </div>
    </div>
  )
}

export default Dropdown
