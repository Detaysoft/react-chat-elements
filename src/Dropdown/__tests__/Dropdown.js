import React, { Component } from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Dropdown from '../Dropdown'

describe('Dropdown component', () => {
  it('should render without issues', () => {
    const component = shallow(<Dropdown items={[]} onSelect={() => {}} />)
    expect(component.length).toBe(1)
    expect(toJson(component)).toMatchSnapshot()
  })
})
