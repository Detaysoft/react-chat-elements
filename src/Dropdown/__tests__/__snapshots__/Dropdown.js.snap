// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should render without issues 1`] = `
<div
  className="rce-dropdown-container"
  onBlur={[Function]}
>
  <Button
    backgroundColor="#3979aa"
    buttonRef={null}
    className={null}
    color="white"
    disabled={false}
    icon={null}
    onClick={[Function]}
    text=""
    type={null}
  />
  <div
    className="rce-dropdown default rce-dropdown-open__nortwest"
  >
    <ul />
  </div>
</div>
`;
